tests/Feature/RoomTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\Room;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Testing\Fluent\AssertableJson;
+use Tests\TestCase;
+
+class RoomTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    public function test_getAllRooms_success(): void
+    {
+        Room::factory()->create();
+
+        $response = $this->getJson('/api/rooms');
+
+        $response->assertStatus(200)
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['message', 'success', 'data'])
+                    ->has('data', 1, function (AssertableJson $data) {
+                    $data->hasAll(['id', 'name', 'rate', 'image', 'min_capacity', 'max_capacity',
+                        'description', 'type_id', 'created_at', 'updated_at'])
```

Comment:
This will need updating - removing type_id, created_at and updated_at. And then adding in type